Potentially Flaky test: CompactionJobQueuesTest.testGetAsync()

**Test name(s)**
`CompactionJobQueuesTest.testGetAsync()`

**Describe the failure observed**
GitHub actions failed on 94167151e2de1be87ac60e25751edf1b557d3a3b for this test. Thought it might have been broken by this commit, but ran locally and was not able to reproduce a failure. Reran the GH action and it passed. Failed here:
https://github.com/apache/accumulo/blob/94167151e2de1be87ac60e25751edf1b557d3a3b/server/manager/src/test/java/org/apache/accumulo/manager/compaction/queue/CompactionJobQueuesTest.java#L402

**Testing Environment:**
4.0/main

I just saw this happen for the first time locally during a build:

```
[ERROR] Tests run: 4, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 4.079 s <<< FAILURE! -- in org.apache.accumulo.manager.compaction.queue.CompactionJobQueuesTest
[ERROR] org.apache.accumulo.manager.compaction.queue.CompactionJobQueuesTest.testGetAsync -- Time elapsed: 1.647 s <<< FAILURE!
org.opentest4j.AssertionFailedError: expected: <true> but was: <false>
	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
	at org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
	at org.junit.jupiter.api.AssertTrue.failNotTrue(AssertTrue.java:63)
	at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:36)
	at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:31)
	at org.junit.jupiter.api.Assertions.assertTrue(Assertions.java:183)
	at org.apache.accumulo.manager.compaction.queue.CompactionJobQueuesTest.testGetAsync(CompactionJobQueuesTest.java:402)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
```

--- a/server/manager/src/test/java/org/apache/accumulo/manager/compaction/queue/CompactionJobQueuesTest.java
+++ b/server/manager/src/test/java/org/apache/accumulo/manager/compaction/queue/CompactionJobQueuesTest.java
@@ -20,7 +20,9 @@ package org.apache.accumulo.manager.compaction.queue;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
 import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 import java.net.URI;
@@ -29,9 +31,11 @@ import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
 import java.util.Optional;
+import java.util.concurrent.ExecutionException;
 import java.util.concurrent.Executors;
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.stream.Stream;
 
@@ -46,7 +50,6 @@ import org.apache.accumulo.core.metadata.schema.TabletMetadata;
 import org.apache.accumulo.core.spi.compaction.CompactionJob;
 import org.apache.accumulo.core.spi.compaction.CompactionKind;
 import org.apache.accumulo.core.spi.compaction.CompactorGroupId;
-import org.apache.accumulo.core.util.UtilWaitThread;
 import org.apache.accumulo.core.util.compaction.CompactionJobImpl;
 import org.apache.hadoop.io.Text;
 import org.junit.jupiter.api.Test;
@@ -393,8 +396,10 @@ public class CompactionJobQueuesTest {
     var future6 = jobQueues.getAsync(cg1);
     assertFalse(future6.isDone());
     future6.orTimeout(10, TimeUnit.MILLISECONDS);
-    // sleep for 20 millis, this should cause future6 to be timed out
-    UtilWaitThread.sleep(20);
+    // Wait for future6 to timeout to make sure future7 will
+    // receive the job when added to the queue
+    var ex = assertThrows(ExecutionException.class, future6::get);
+    assertInstanceOf(TimeoutException.class, ex.getCause());
     var future7 = jobQueues.getAsync(cg1);
     assertFalse(future7.isDone());
     // since future5 was canceled and future6 timed out, this addition should go to future7

